client/core/core.go

```diff
@@ -6324,80 +6323,82 @@ func (c *Core) dbTrackers(dc *dexConnection) (map[order.OrderID]*trackedTrade, e
-// unlocked. To prevent spamming wallet connections, the 'failed' map will be
-// populated with asset IDs for which the attempt to connect or unlock has
-// failed. The failed map should be passed on subsequent calls for other dexes.
-func (c *Core) loadDBTrades(dc *dexConnection, crypter encrypt.Crypter, failed map[uint32]struct{}) ([]*trackedTrade, error) {
-	// Parse the active trades and see if any wallets need unlocking.
-	trades, err := c.dbTrackers(dc)
+// loadDBTrades load's the active trades from the db, populates the trade's
+// wallets field and some other metadata, and adds the trade to the
+// dexConnection's trades map. readyToTick will still be set to false, so
+// resumeTrades must be run before the trades will be processed.
+func (c *Core) loadDBTrades(dc *dexConnection) error {
+	trackers, err := c.dbTrackers(dc)
 	if err != nil {
-		return nil, fmt.Errorf("error retrieving active matches: %w", err)
+		return fmt.Errorf("error retrieving active matches: %w", err)
 	}
 
 	errs := newErrorSet(dc.acct.host + ": ")
-	ready := make([]*trackedTrade, 0, len(trades))
-	for _, trade := range trades {
-		if !trade.isActive() {
+	var tradesLoaded uint32
+	for _, tracker := range trackers {
+		if !tracker.isActive() {
 			// In this event, there is a discrepancy between the active criteria
 			// between dbTrackers and isActive that should be resolved.
-			c.log.Warnf("Loaded inactive trade %v from the DB.", trade.ID())
+			c.log.Warnf("Loaded inactive trade %v from the DB.", tracker.ID())
 			continue
 		}
-		base, quote := trade.Base(), trade.Quote()
-		_, baseFailed := failed[base]
-		_, quoteFailed := failed[quote]
-		if !baseFailed {
-			baseWallet, err := c.connectedWallet(base)
-			if err != nil {
-				baseFailed = true
-				failed[base] = struct{}{}
-				c.log.Errorf("Connecting to wallet %s failed: %v", unbip(base), err)
-			} else if !baseWallet.unlocked() {
-				err = baseWallet.Unlock(crypter)
-				if err != nil {
-					baseFailed = true
-					failed[base] = struct{}{}
-					c.log.Errorf("Unlock wallet %s failed: %v", unbip(base), err)
-				}
-			}
+
+		trade := tracker.Trade()
+
+		walletSet, assetConfigs, versCompat, err := c.walletSet(dc, tracker.Base(), tracker.Quote(), trade.Sell)
+		if err != nil {
+			errs.add("failed to load wallets for trade ID %s: %w", tracker.ID(), err)
+			continue
 		}
-		if !baseFailed && !quoteFailed {
-			quoteWallet, err := c.connectedWallet(quote)
-			if err != nil {
-				quoteFailed = true
-				failed[quote] = struct{}{}
-				c.log.Errorf("Connecting to wallet %s failed: %v", unbip(quote), err)
-			} else if !quoteWallet.unlocked() {
-				err = quoteWallet.Unlock(crypter)
-				if err != nil {
-					quoteFailed = true
-					failed[quote] = struct{}{}
-					c.log.Errorf("Unlock wallet %s failed: %v", unbip(quote), err)
-				}
+
+		tracker.wallets = walletSet
+		dc.tradeMtx.Lock()
+		if _, found := dc.trades[tracker.ID()]; found {
+			dc.tradeMtx.Unlock()
+			return nil
```

Comment:
`Logout()` won't work if there are active deals in the map. I don't see any way this will get hit currently unless there is a bug causing duplicates to be returned from the db, but I 100% agree the check should be there. As you say, `loadDBTrades` could be called differently in the future.